lib/std/syncio.nim

```diff
@@ -246,7 +246,7 @@ when defined(windows):
     # machine. We also enable `setConsoleOutputCP(65001)` now by default.
     # But we cannot call printf directly as the string might contain \0.
     # So we have to loop over all the sections separated by potential \0s.
-    var i = c_fprintf(f, "%s", s)
+    var i = int c_fprintf(f, "%s", s)
```

Comment:
won't using int mean that this will still fail on 32-bit systems for big files?

Reply:
Nope. `s.len` returns a `int` type, too and `i` is equal to `s.len`. It cannot exceed 2GB on 32-bit systems.

Reply:
Did I help?

Reply:
You helped a bunch! Thank you!